File: minimaratonas/1ºciclo/todosProblemas/corrida/Corrida/sol.cpp
#include<bits/stdc++.h>
using namespace std;


int v[1000010];
int main(){
	int n; scanf("%d", &n);
	
	for(int i=0;i<n;i++){
		scanf("%d", &v[i+1]);
	}
	
	int a, b;
	scanf("%d %d", &a, &b);
	
	int menor = v[a];
	int maior = v[b];
	for(int i=a;i<=b;i++){
		menor = min(menor, v[i]);
		maior = max(maior, v[i]);
	}
	
	printf("%d\n", maior-menor);
}

File: minimaratonas/1ºciclo/todosProblemas/corrida/sol.py
n = int(input())
v = input().split()

for i in range(0, len(v)):
    v[i] = int(v[i])
    

l, r = input().split()
l = int(l)
r = int(r)

menor = v[l-1]
maior = v[l-1]

for i in range(l-1, r):
    menor = min(menor, v[i])
    maior = max(maior, v[i])
    
    
print(maior-menor)    

